Source/Core/Core/Analytics.cpp

```diff
@@ -247,12 +247,26 @@ void DolphinAnalytics::MakePerGameBuilder()
 	builder.AddData("movie", Movie::IsMovieActive());
 
 	// Controller information
+	// We grab enough to tell what percentage of our users are playing with keyboard/mouse, some kind of gamepad
+	// or the official gamecube adapter.
 	builder.AddData("gcadapter-detected", GCAdapter::IsDetected());
 
-	// For privacy reasons, limit this to type of the first controller.
+	bool hasController = true;
 	// The ControllersNeedToBeCreated() check is enough to ensure GetController(0) won't return nullptr or throw exceptions.
 	if (!Pad::GetConfig()->ControllersNeedToBeCreated())
-		builder.AddData("controller-type", Pad::GetConfig()->GetController(0)->default_device.name);
+	{
+		const auto& controller = Pad::GetConfig()->GetController(0)->default_device;
```

Comment:
This added code should likely be in it's own local function that returns a boolean. This sanitizing has nothing to do with the creation of the builder directly.